### PR TITLE
BigQuery Authorized datasets

### PR DESCRIPTION
**Pull request by slevenick**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/10555

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
bigquery: added support for authorized datasets to `google_bigquery_dataset.access` and `google_bigquery_dataset_access`
```

### CODE REVIEW

File: mmv1/products/bigquery/api.yaml [hunk 1]
```diff
@@ -112,6 +112,31 @@ objects:
                     A-Z), numbers (0-9), or underscores (_). The maximum length
                     is 1,024 characters.
                   required: true
+            - !ruby/object:Api::Type::NestedObject
+              name: 'dataset'
+              description: |
+                Grants all resources of particular types in a particular dataset read access to the current dataset.
+              properties:
+                - !ruby/object:Api::Type::NestedObject
+                  name: 'dataset'
```
Comment (rileykarson) [inline mmv1/products/bigquery/api.yaml:121]: Is this one required? It feels like it should beat a glance.

File: mmv1/products/bigquery/api.yaml [hunk 2]
```diff
@@ -377,6 +409,39 @@ objects:
...
+          - user_by_email
+          - group_by_email
+          - domain
+          - special_group
+          - iam_member
+          - view
+          - dataset
+        properties:
+          - !ruby/object:Api::Type::NestedObject
+            name: 'dataset'
```
Comment (rileykarson) [inline mmv1/products/bigquery/api.yaml:426]: Same deal- should this be required?

File: mmv1/products/bigquery/api.yaml [hunk 3]
```diff
@@ -248,6 +273,7 @@ objects:
       - specialGroup
       - iamMember
       - view
+      - dataset
```
Comment (rileykarson) [inline mmv1/products/bigquery/api.yaml:276]: Changing an identity- spooky! That one's been a while. VCR thinks it's safe, so seems fine though.

**Review by rileykarson — APPROVED** [review]

File: mmv1/products/bigquery/api.yaml (hunk 1 above)
Comment (slevenick) [reply mmv1/products/bigquery/api.yaml:121]: Yup!

File: mmv1/products/bigquery/api.yaml (hunk 2 above)
Comment (slevenick) [reply mmv1/products/bigquery/api.yaml:427]: Yup, good catch!

**Pull request merged**